azurelinuxagent/ga/exthandlers.py

```diff
@@ -1191,6 +1249,17 @@ def read_output():
 
         return read_output()
 
+    def load_handler_configuration(self):
+        config_file = self.get_handler_configuration_file()
+        try:
+            data = json.loads(fileutil.read_file(config_file))
```

Comment:
we should validate the file when we read it...

Reply:
The Validation is happening within the Class creation. The new `load_handler_configuration` method is self-contained.

```
        try:
            handler_config = HandlerConfiguration(data)
        except ExtensionError as e:
            self.logger.info('Failed to load resource manifest file: {0}'.format(ustr(e)))
```